samples/property/provider/src/main.rs

```diff
@@ -145,6 +121,63 @@ fn start_ambient_air_temperature_data_stream(subscription_map: Arc<Mutex<Subscri
     });
 }
 
+/// Publish a message to a MQTT broker located.
+///
+/// # Arguments
+/// `host` - The MQTT broker's URI.
+/// `topic` - The topic to publish to.
+/// `content` - The message to publish.
+fn publish_message(host: &str, topic: &str, content: &str) -> Result<(), String> {
+    let create_opts = mqtt::CreateOptionsBuilder::new()
+        .server_uri(host)
+        .client_id(MQTT_CLIENT_ID.to_string())
+        .finalize();
+
+    let client = mqtt::Client::new(create_opts)
+        .map_err(|err| format!("Failed to create the client due to '{:?}'", err))?;
+
+    let conn_opts = mqtt::ConnectOptionsBuilder::new()
+        .keep_alive_interval(Duration::from_secs(30))
+        .clean_session(true)
+        .finalize();
+
+    let _connect_response =
+        client.connect(conn_opts).map_err(|err| format!("Failed to connect due to '{:?}", err));
+
+    let msg = mqtt::Message::new(topic, content, MQTT_QOS);
+    if let Err(err) = client.publish(msg) {
+        return Err(format!("Failed to publish message due to '{:?}", err));
+    }
+
+    let tok = client.disconnect(None);
+    tok.unwrap();
+
+    Ok(())
+}
+
+/// Convert a DTMI to a MQTT topic name.
+/// The conversion will strip off the scheme (i.e. the "dtmi:" prefix)
+/// and replace all seprators (':' and ';') with a slash.
+///
+/// # Arguments
+/// `dtmi` - The DTMI.
+fn convert_dtmi_to_topic(dtmi: &str) -> Result<String, String> {
```

Comment:
You can reduce this function by doing the following:
```
fn convert_dtmi_to_topic(dtmi: &str) -> Result<String, String> {
    const DTMI_COLON_SCHEME: &str = "dtmi:";
    if !dtmi.starts_with(DTMI_COLON_SCHEME) {
        return Err("Invalid dtmi".to_string());
    }
     // Replaces ":" and ";" in a string, after the "dtmi:", with "/"
    let topic = dtmi[DTMI_COLON_SCHEME.len()..].replace(&[':', ';'][..], "/");
    Ok(topic)
}
```

Reply:
But I also remove the "dtmi:" from the front of the dtmi param.

Reply:
https://doc.rust-lang.org/std/primitive.str.html#method.strip_prefix Maybe this could be used to strip the `dtmi/` after replace is run?

Reply:
It could, but the complexity will be similar to the original one.  I'm going to leave it as is.